src/services/funcService.ts

```diff
@@ -0,0 +1,132 @@
+import yaml from "js-yaml";
+import rimraf from "rimraf";
+import Serverless from "serverless";
+import httpBinding from "../plugins/func/bindingTemplates/http.json";
+import { BaseService } from "./baseService";
+import fs from "fs";
+
+export class FuncService extends BaseService {
+  public constructor(serverless: Serverless, options: Serverless.Options){
+    super(serverless, options, false);
+  }
+
+  public add() {
+    const functionName = this.options["name"];
+    if (!functionName) {
+      this.log("Need to provide a name of function to add");
+      return;
+    }
+    if (this.exists(functionName)) {
+      this.serverless.cli.log(`Function ${functionName} already exists`);
+      return;
+    }
+    this.createHandler(functionName);
+    this.addToServerlessYml(functionName);
+  }
+
+  public remove() {
+    const functionName = this.options["name"];
+    if (!functionName) {
+      this.log("Need to provide a name of function to remove");
+      return;
+    }
+    if (!this.exists(functionName)) {
+      this.log(`Function ${functionName} does not exist`);
+      return;
+    }
+    const fileName = `${functionName}.js`;
+    if (fs.existsSync(fileName)) {
+      fs.unlinkSync(fileName);
+    }
+    if (fs.existsSync(functionName)) {
+      rimraf.sync(functionName);
+    }
+    this.removeFromServerlessYml(functionName);
+  }
+
+  private exists(functionName: string) {
+    return (functionName in this.slsFunctions());
+  }
+
+  private createHandler(functionName: string) {
+    this.serverless.utils.writeFileSync(`./${functionName}.js`, this.getFunctionHandler(functionName))
+  }
+
+  private addToServerlessYml(functionName: string) {
+    const functions = this.slsFunctions();
+    functions[functionName] = this.getFunctionSlsObject(functionName)
+    this.updateFunctionsYml(functions)
+  }
+
+  private removeFromServerlessYml(functionName: string) {
+    const functions = this.slsFunctions();
+    delete functions[functionName];
+    this.updateFunctionsYml(functions)
+  }
+
+  private getServerlessYml() {
+    return this.serverless.utils.readFileSync("serverless.yml");
```

Comment:
Does the framework have a way to get the name of the yaml file? With the new `--config` option, it's possible that the file won't be named `serverless.yml`. We don't have to over-index around this, but just something to keep in mind.

Reply:
Good point. Wasn't accounting for this. I'll investigate further here.

Reply:
Added a getter that defaults to `serverless.yml`. Good callout